builtin/logical/aws/secret_access_keys.go

```diff
@@ -123,6 +173,16 @@ func (b *backend) secretAccessKeysRenew(
 func secretAccessKeysRevoke(
 	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
 	// Get the username from the internal data
+	isSTS, ok := req.Secret.InternalData["is_sts"]
+	if !ok {
```

Comment:
This will break upgrades -- existing secrets won't be able to revoke. This needs to gracefully handle the lack of `is_sts` by assuming IAM (e.g. just forcing `isSTS` to `false`)